ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java

```diff
@@ -114,16 +115,23 @@ public DescribeConnectorExecutor() {
       sources = ImmutableList.of();
     }
 
-    List<KsqlWarning> warnings;
-    List<String> topics;
-    try {
-      topics = getTopics(serviceContext.getAdminClient(), connector);
-      warnings = ImmutableList.of();
-    } catch (final Exception e) {
+    final ConnectResponse<Map<String, Map<String, List<String>>>> topicsResponse = serviceContext
+        .getConnectClient()
+        .topics(connectorName);
+
+    final List<KsqlWarning> warnings;
+    final List<String> topics;
```

Comment:
we should use the information here before getting the data sources above (basically instead of calling `matches` on L106 above we should check `topics#contains`)

Reply:
Moved up with a small optimization to avoid hitting the `topics` endpoint if the first REST call did not return the connector's info. But if the doubly nested if does look great or you think it's worth calling regardless I can change the if/else